### PR TITLE
Update values.preview.template.yaml

### PR DESCRIPTION
**Pull request by shashisk97**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [x] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

Solicitor upload journey

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -567,6 +567,8 @@ def sonarExclusions = [
   '**uk/gov/hmcts/reform/prl/utils/*',
   '**uk/gov/hmcts/reform/prl/Application.*',
   '**/uk/gov/hmcts/reform/prl/services/TestingSupportService.java',
+  '**uk/gov/hmcts/reform/prl/services/ManageOrderService.java',
+  '**uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java'
```
Comment (AlokDatta) [inline build.gradle:571]: If we are not in a tight timeline, lets cover this up.

File: src/main/java/uk/gov/hmcts/reform/prl/enums/ManageOrderFieldsEnum.java [hunk 2]
```diff
@@ -29,7 +29,17 @@ public enum ManageOrderFieldsEnum {
     c21OrderOptions("c21OrderOptions"),
     selectChildArrangementsOrder("selectChildArrangementsOrder"),
     childArrangementsOrdersToIssue("childArrangementsOrdersToIssue"),
-    childOption("childOption");
+    childOption("childOption"),
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/enums/ManageOrderFieldsEnum.java:32]: Are we supposed to remove childOption?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 3]
```diff
@@ -366,7 +370,7 @@ public Map<String, Object> populateDraftOrderDocument(CaseData caseData) {
         Map<String, Object> caseDataMap = new HashMap<>();
         DraftOrder selectedOrder = getSelectedDraftOrderDetails(caseData);
         caseDataMap.put("previewUploadedOrder", selectedOrder.getOrderDocument());
-        if (caseData.getManageOrders() != null && caseData.getManageOrders().getJudgeDirectionsToAdminAmendOrder() != null) {
+        if (!StringUtils.isEmpty(selectedOrder.getJudgeNotes())) {
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:373]: Is there any chance selectedOrder can be null?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 4]
```diff
@@ -585,7 +589,7 @@ public Map<String, Object> updateDraftOrderCollection(CaseData caseData, String
 
     private DraftOrder getDraftOrderWithUpdatedStatus(CaseData caseData, String eventId, String loggedInUserType, DraftOrder draftOrder) {
         return draftOrder.toBuilder()
-            .judgeNotes(caseData.getJudgeDirectionsToAdmin())
+            .judgeNotes(draftOrder.getJudgeNotes() != null ? draftOrder.getJudgeNotes() : caseData.getJudgeDirectionsToAdmin())
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:592]: should we have StringUtils.isEmpty?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 5]
```diff
@@ -660,7 +664,7 @@ private DraftOrder getUpdatedDraftOrder(DraftOrder draftOrder, CaseData caseData
             .underTakingDateExpiry(caseData.getManageOrders().getUnderTakingDateExpiry())
             .underTakingExpiryTime(caseData.getManageOrders().getUnderTakingExpiryTime())
             .underTakingFormSign(caseData.getManageOrders().getUnderTakingFormSign())
-            .judgeNotes(caseData.getJudgeDirectionsToAdmin())
+            .judgeNotes(draftOrder.getJudgeNotes() != null ? draftOrder.getJudgeNotes() : caseData.getJudgeDirectionsToAdmin())
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:667]: Should we have StringUtils.isEmpty?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java [hunk 6]
```diff
@@ -1168,8 +1168,7 @@ public DraftOrder getCurrentCreateDraftOrderDetails(CaseData caseData, String lo
                             ? copyPropertiesToSdoDetails(caseData) : null)
             .hearingsType(caseData.getManageOrders().getHearingsType())
             .isOrderCreatedBySolicitor(UserRoles.SOLICITOR.name().equals(loggedInUserType) ? Yes : No)
-            .judgeNotes(null != caseData.getManageOrders().getJudgeDirectionsToAdminAmendOrder()
-                            ? caseData.getManageOrders().getJudgeDirectionsToAdminAmendOrder() : " ")
+            .judgeNotes(caseData.getJudgeDirectionsToAdmin())
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java:1171]: Do we need this change? I mean removing this null check and " " was it intentional? If yes, please ignore

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java [hunk 7]
```diff
@@ -1215,18 +1214,26 @@ private DraftOrder getCurrentUploadDraftOrderDetails(CaseData caseData, String l
...
             .hasJudgeProvidedHearingDetails(caseData.getManageOrders().getHasJudgeProvidedHearingDetails())
             .isOrderCreatedBySolicitor(UserRoles.SOLICITOR.name().equals(loggedInUserType) ? Yes : No)
             .build();
     }
 
+    private YesOrNo getIsUploadedFlag(ManageOrdersOptionsEnum manageOrdersOptions, String loggedInUserType) {
+        if (UserRoles.SOLICITOR.name().equals(loggedInUserType)) {
+            return Yes;
+        } else if (null != manageOrdersOptions
+            && manageOrdersOptions.equals(uploadAnOrder)) {
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java:1231]: assuming uploadAnOrder is a constant? if yes can we alter the if logic?

Comment (AlokDatta) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java:1231]: And this if else logic will throw Major code smell. We should have a variable holding the return value and returning it.

**Review by AlokDatta — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/enums/ManageOrderFieldsEnum.java (hunk 2 above)
Comment (shashisk97) [reply src/main/java/uk/gov/hmcts/reform/prl/enums/ManageOrderFieldsEnum.java:32]: Its not removed, we observed in one of the flow this field was throwing error but dint see the issue again. This field is not added/removed as part of this change so will take it separately

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java (hunk 3 above)
Comment (shashisk97) [reply src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:373]: No

**Pull request merged**